### PR TITLE
add location type filter

### PR DESCRIPTION
**Pull request by jamesgorrie**
After much tooing-and-frowing - it seems like this is the much easier way to do this.

We will be adding some more data on ClosedStores / OpenShelves later.

### CODE REVIEW

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/FiltersAndAggregationsBuilder.scala [hunk 1]
```diff
@@ -79,18 +79,19 @@ class FiltersAndAggregationsBuilder(
...
-    case VisibleWorkFilter         => None
-    case _: LanguageFilter         => Some(AggregationRequest.Language)
-    case _: GenreFilter            => Some(AggregationRequest.Genre)
-    case _: SubjectFilter          => Some(AggregationRequest.Subject)
-    case _: LicenseFilter          => Some(AggregationRequest.License)
-    case _: IdentifiersFilter      => None
-    case _: CollectionPathFilter   => None
-    case _: CollectionDepthFilter  => None
-    case _: AccessStatusFilter     => None
+    case _: FormatFilter             => Some(AggregationRequest.Format)
```
Comment (jamieparkinson) [inline api/api/src/main/scala/uk/ac/wellcome/platform/api/services/FiltersAndAggregationsBuilder.scala:82]: Is there a meaningful change in ordering here? Can't see anything but sort of hard to read

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DocumentFilter.scala [hunk 2]
```diff
@@ -1,14 +1,18 @@
 package uk.ac.wellcome.platform.api.models
 
 import uk.ac.wellcome.models.work.internal.AccessStatus
-
 import java.time.LocalDate
 
+import uk.ac.wellcome.display.models.LocationTypeRequest
```
Comment (jamieparkinson) [inline api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DocumentFilter.scala:6]: Not totally sure about the `Request` naming... but can't think of anything better.

**Review by jamieparkinson — APPROVED** [review]

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/FiltersAndAggregationsBuilder.scala (hunk 1 above)
Comment (jamesgorrie) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/services/FiltersAndAggregationsBuilder.scala:82]: Put the item bits together, and the tabbing changed

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DocumentFilter.scala (hunk 2 above)
Comment (jamesgorrie) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DocumentFilter.scala:6]: We've used it elsewhere, so copying. I think I came up with it and felt weird at the time. Query?

Comment (jamieparkinson) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DocumentFilter.scala:6]: Yeah think I prefer query!

**Pull request merged**